### ISSUE TITLE
findFirst method return null, if record is not found

### ISSUE THREAD

**Issue by Druid33** [phalcon/cphalcon#15077]
Hi,
i found that in phalcon4, when record is not found with findfirst method, this method retrun null, not false, like in phalcon3.
In offcial phalcon4 documentation its not mentioned. But if someoone looks in to phalcon .zep files, its ease to find, that findFirst can return mixed data type (because of Pahlcon/Mvc/Model/Query->execute() can https://github.com/phalcon/cphalcon/blob/fe17e3ae897b8d3f41d65ed328fcb734d1955b51/phalcon/Mvc/Model/Query.zep#L3643).

https://github.com/phalcon/ide-stubs/blob/d2adae3fba488ccba913a3f7fc310f23aa6de4a3/src/Phalcon/Mvc/Model.php#L791

**Comment by niden** [comment]
> i found that in phalcon4, when record is not found with findfirst method, this method retrun null, not false, like in phalcon3.
> In offcial phalcon4 documentation its not mentioned. But if someoone looks in to phalcon .zep 

This probably needs to be moved to the stubs repo. The docs actually do state the change.

https://docs.phalcon.io/4.0/en/db-models

A but further down where the `public static function findFirst()....` is mentioned there is a callout that shows the change. Additionally the change is reflected in the `upgrade` document:

https://docs.phalcon.io/4.0/en/upgrade#mvcmodel

**Comment by Jeckerson** [comment]
This is not a bug, but a new feature of `v4`. Which make more sense in PHP7.

Example:
```
public function someMethod(): ?SomeInterface // where '?' is null return type
```

**Comment by niden** [comment]
This has been resolved in #15080 @Jeckerson we forgot the entry in the changelog.

**Comment by Jeckerson** [comment]
Fixed in `4.1.x` branch.
